
---
title: "ITS2_algal_symbionts"
output: html_notebook
editor_options: 
  chunk_output_type: console
chunk_output_type: console
---
This is from the Microbial Landscape experiment (summer 2019). ITS2 data from Symportal pipeline. 
Last updated:20220622

* load in all data. Use the raw reads to view seqs and then create rarefied relative abundance df yourself
```{r}
#rm(list=ls())

library(tidyverse)
library(readxl)
library(phyloseq)
library(janitor)
library("writexl")

setwd("~/Desktop/Microbial landscape/R/Mcapitata_microbial_landscape/ITS2_Summer_symportal2020")
```
Reading in data with Ross code, this is relA. 
# Relative Abundance 
Read in coral ITS2 profiles: "coral"
```{r}  
#add metadata to symportal submission data
Symp_sub<-readxl::read_xlsx("MCL_SymPortal_submission_input.xlsx", skip = 1)
Metadata<-read.csv("MCL19_metadata_20220621.csv")
Metadata<-rename(Metadata, sample_name=UNIQUEID) #rename UNIQUEID column to sample_name
Symp_sub_Meta<-left_join(Symp_sub, Metadata, by="sample_name") # merge
#save in correct format:
write_xlsx(Symp_sub_Meta,"MCL_Symp_Metadata_20220621.xlsx")

sam0 <- readxl::read_xlsx("MCL_Symp_Metadata_20220621.xlsx") #Symportal metadata combined from above
sam1 <- as.matrix(sam0[, -1])
rownames(sam1) <- sam0$sample_name
sam <- sample_data(data.frame(sam1))

tax0 <- read_tsv(
  file  = "20201214_matsuda_MCL/its2_type_profiles/133_20201216_DBV_20201216T011417.profiles.relative.abund_and_meta.txt",
  n_max = 6) %>%
  dplyr::select(-2) %>% 
  gather(UID, value, -1) %>% 
  spread(1, value) %>%
  clean_names()

tax1 <- as.matrix(tax0[, -1], dimnames = list(tax0$uid, colnames(tax0[-1])))
rownames(tax1) <- tax0$uid
tax <- tax_table(tax1)

otu0 <- read_tsv(
  file  = "20201214_matsuda_MCL/its2_type_profiles/133_20201216_DBV_20201216T011417.profiles.relative.abund_and_meta.txt") %>% 
  rename(sample_name = ...2) %>% #rename column "...2", previously "X2" in symportal language
  select(-1) %>%
  slice(7:n()) %>%
  mutate_at(2:ncol(.), as.numeric)
otu1 <- as.matrix(otu0[, -1])
rownames(otu1) <- otu0$sample_name
otu <- otu_table(otu1, taxa_are_rows = FALSE)

coral <- phyloseq(otu, tax, sam)
 
```
Read in coral post-QC sequence variants: "coralDIV"
```{r}  
sam0 <- read_xlsx("MCL_Symp_Metadata_20220621.xlsx")
sam1 <- as.matrix(sam0[, -1])
rownames(sam1) <- sam0$sample_name
sam <- sample_data(data.frame(sam1))


taxnames <- read_tsv(
  file  = "20201214_matsuda_MCL/post_med_seqs/133_20201216_DBV_20201216T011417.seqs.relative.abund_only.txt",
  n_max = 0) %>%
  select(-1) %>%
  names(.)
tax0 <- data_frame(
  DIV = taxnames,
  clade = str_extract(DIV, "[A-Z]")
)
tax1 <- as.matrix(tax0)
rownames(tax1) <- tax0$DIV
tax <- tax_table(tax1)
otu0 <- read_tsv(
  file  = "20201214_matsuda_MCL/post_med_seqs/133_20201216_DBV_20201216T011417.seqs.relative.abund_and_meta.txt") %>%
  select(-1, -(3:33))
otu1 <- as.matrix(otu0[, -1])
rownames(otu1) <- otu0$sample_name
otu <- otu_table(otu1, taxa_are_rows = FALSE)
coralDIV <- phyloseq(otu, tax, sam)
```

```{r} 
save(coral, coralDIV, file = "data/coral_phyloseq_2022.RData")
```

#Reading in data by ABSOLUTE abundances
This is the same as above just by absolute (you may not even want to keep this)

Profiles Absolute
```{r}
 
sam0 <- readxl::read_xlsx("MCL_Symp_Metadata_20220621.xlsx") #Symportal metadata combined from above
# make all factors
cols <- c("sample_type", "TagID", "Colony.Color", "Samp.Date","Land_Ocean","SampLoc","BranchLoc","Zoox","Location")
sam0[cols] <- lapply(sam0[cols], factor)  ## as.factor() could also be used
sam1 <- as.matrix(sam0[, -1])
rownames(sam1) <- sam0$sample_name
sam <- sample_data(data.frame(sam1))

tax0 <- read_tsv(
  file  = "20201214_matsuda_MCL/its2_type_profiles/133_20201216_DBV_20201216T011417.profiles.absolute.abund_and_meta.txt",
  n_max = 6) %>%
  dplyr::select(-2) %>% 
  gather(UID, value, -1) %>% 
  spread(1, value) %>%
  clean_names()

tax1 <- as.matrix(tax0[, -1], dimnames = list(tax0$uid, colnames(tax0[-1])))
rownames(tax1) <- tax0$uid
tax <- tax_table(tax1)

otu0 <- read_tsv(
  file  = "20201214_matsuda_MCL/its2_type_profiles/133_20201216_DBV_20201216T011417.profiles.absolute.abund_and_meta.txt") %>% 
  rename(sample_name = ...2) %>%
  select(-1) %>%
  slice(7:n()) %>%
  mutate_at(2:ncol(.), as.numeric)
otu1 <- as.matrix(otu0[, -1])
rownames(otu1) <- otu0$sample_name
otu <- otu_table(otu1, taxa_are_rows = FALSE)

coral_absolutePro <- phyloseq(otu, tax, sam)
```
DIVs absolute
```{r} 

taxnames <- read_tsv(
  file  = "20201214_matsuda_MCL/post_med_seqs/133_20201216_DBV_20201216T011417.seqs.absolute.abund_only.txt",
  n_max = 0) %>%
  select(-1) %>%
  names(.)
tax0 <- data_frame(
  DIV = taxnames,
  clade = str_extract(DIV, "[A-Z]")
)
tax1 <- as.matrix(tax0)
rownames(tax1) <- tax0$DIV
tax <- tax_table(tax1)
otu0 <- read_tsv(
  file  = "20201214_matsuda_MCL/post_med_seqs/133_20201216_DBV_20201216T011417.seqs.absolute.abund_and_meta.txt") %>%
  select(-1, -(3:33))
otu1 <- as.matrix(otu0[, -1])
rownames(otu1) <- otu0$sample_name
otu <- otu_table(otu1, taxa_are_rows = FALSE)
coral_Absolute_DIV <- phyloseq(otu, tax, sam)
```
Save as an obj
```{r}
save(coral_absolutePro, coral_Absolute_DIV, file = "data/Absolute_coral_phyloseq2022.RData")
```

#Analysis (RCunning Script) REL Abundance -  
```{r} 
library(tidyverse)
library(janitor)
library(phyloseq)  # source('http://bioconductor.org/biocLite.R'); biocLite('phyloseq')
library(zoo)
library(stringr)
library(vegan)
library(multcompView)
library("ggpubr")

# Load data (see data_exploration.Rmd)
load("data/coral_phyloseq_2022.RData") # Rel A data
load("data/Absolute_coral_phyloseq2022.RData") #absolute reads data

```
Clean -dont run this yet for Raw
```{r}
coral2 <- coral %>% 
  subset_samples(!is.na(host_species))
coralDIV2 <- coralDIV %>% 
  subset_samples(!is.na(host_species))

# HERE YOU MAKE RELA DATA ACTUALLY OUT OF 100% 
# Transform to relative abundance
coral2 <- transform_sample_counts(coral2, function(x) 100 * x/sum(x))
coralDIV2 <- transform_sample_counts(coralDIV2, function(x) 100 * x/sum(x))

```

Data wrangle: remove samples with under 5k 
```{r} 
#lets see what is up with the low samps

LowSamps<-readxl::read_xlsx("20201214_matsuda_MCL/post_med_seqs/133_20201216_DBV_20201216T011417.seqs.absolute.meta_only.xlsx")

LowSamps$post_taxa_id_absolute_symbiodiniaceae_seqs_log<-log10(LowSamps$post_taxa_id_absolute_symbiodiniaceae_seqs)
hist(LowSamps$post_taxa_id_absolute_symbiodiniaceae_seqs_log, breaks=20)

```
Look at the low samples: no patterns (this isn't running right now, figure it out (June 21, 2022))
```{r} 
sample_names(coral_Absolute_DIV) #print sample names

#look at NTCs
#Negative Controls
  NTCtest <- subset_samples(coral_Absolute_DIV, sample_names =="NTC"| sample_names =="NTC2") #subset multiple samples
   NTCtest.prune <- prune_taxa(taxa_sums(NTCtest) >0, NTCtest)
     plot_bar(NTCtest.prune,fill = "DIV")+
      geom_bar(aes(color=DIV, fill=DIV), stat="identity", position="stack")
     
#Samples under 2000 hits 
LOWtest <- subset_samples(coralDIV2, sample_name =="S6"| sample_name =="G7"|sample_name =="Q11"|sample_name =="G8"|sample_name =="B13"|sample_name =="B6"|sample_name =="C14") #subset multiple samples
   LOWtest.prune <- prune_taxa(taxa_sums(LOWtest) >0, LOWtest) # prune all 0s
     
   plot_bar(LOWtest.prune,fill = "DIV")+                     # plot DIVS
    geom_bar(aes(color=DIV, fill=DIV), stat="identity", position="stack")
     
   plot_bar(LOWtest.prune,fill = "clade")+                     # plot Clades
      geom_bar(aes(color=clade, fill=clade), stat="identity", position="stack")
     
```
 
Remove low samples from the datasets (less than 2000)
coral_absolutePro
coral_Absolute_DIV

```{r} 
#DIVs 
to_remove <- c("NTC", "NTC2", "S6", "G7","Q11","G8", "B13","B6") #make list of samples to remove with less than 2000 seqs
coral_Absolute_DIV2.prune <- prune_samples(!(sample_names(coral_Absolute_DIV) %in% to_remove), coral_Absolute_DIV) #remove
#sample_names(coralDIV2.prune) # check to see they are removed

#Pros
to_remove <- c("NTC", "NTC2", "S6", "G7","Q11","G8", "B13","B6") #make list of samples to remove with less than 2000 seqs
coral_absolutePro.prune <- prune_samples(!(sample_names(coral_absolutePro) %in% to_remove), coral_absolutePro) #remove
#sample_names(coral2.prune) # check to see they are removed

```

# DIVs examine data and clean up
```{r}
## check out data
ntaxa(coral_Absolute_DIV2.prune)  #num taxa
nsamples(coral_Absolute_DIV2.prune)   #num samples
sample_names(coral_Absolute_DIV2.prune)[1:5] #samp names
rank_names(coral_Absolute_DIV2.prune) #ranks are DIV and clade
sample_variables(coral_Absolute_DIV2.prune) #all metadata cats

# create df of sample data to view 
sample.data <- as(sample_data(coral_Absolute_DIV2.prune), "data.frame") #create sample data frame to view
sample.data$LibrarySize <- sample_sums(coral_Absolute_DIV2.prune)
sample.data <- sample.data[order(sample.data$LibrarySize),]
sample.data$Index <- seq(nrow(sample.data))
ggplot(data = sample.data, aes(x=Index, y=LibrarySize, color = sample_type)) +
  geom_point()
```

Making rare and non-rare dataframes
```{r}
#DIVs
#make a non-rarefied df:
#Remove all samples with <2000 reads and create sample data
#These data also useful for beta diversity analyses (non-rarefied)
coral_Absolute_DIV2.prune_nonrare <- prune_samples(sample_sums(coral_Absolute_DIV2.prune)>=2000, coral_Absolute_DIV2.prune) # remove samps <2000, save as new obj
coral_Absolute_DIV2.prune_nonrare_sd <- as(sample_data(coral_Absolute_DIV2.prune_nonrare), "data.frame")




#Make a rarefied phyloseq object for alpha diversity analyses
coral_Absolute_DIV2.rare <- rarefy_even_depth(coral_Absolute_DIV2.prune, sample.size = 2000, rngseed = 711)   #C14 sample removedbc less than 2000 
sample_sums(coral_Absolute_DIV2.rare) #Double check that the sub-sampling worked, this should report 2000 for each sample
coralDIV2.rare <- as(sample_data(coral_Absolute_DIV2.rare), "data.frame")
coralDIV2.rare



#Rareify Profiles/Clades coral_absolutePro.prune
coral_Absolute_Pro.rare <- rarefy_even_depth(coral_absolutePro.prune, sample.size = 2000, rngseed = 711)   #C14 sample removedbc less than 2000 
sample_sums(coral_Absolute_Pro.rare) #Double check that the sub-sampling worked, this should report 2000 for each sample
coralPro.rare <- as(sample_data(coral_Absolute_Pro.rare), "data.frame")
coralPro.rare
```

Turn rarified df in to relative abundance: coral_Absolute_DIV2.rare
```{r}
#RELA 
DIV_RelA  = transform_sample_counts(coral_Absolute_DIV2.rare, function(x) x / sum(x) )  #save as RELA DIVs
data.ra <- as(sample_data(DIV_RelA), "data.frame") #look at samp data
#write.csv(data.ra, "DIV_its2_sampledata.csv")

   DIV_RelA = prune_taxa(taxa_sums(DIV_RelA) > 0, DIV_RelA) #this removes any OTU with 0s

```

# make distance matrix
(Code from TT17)
```{r}
#make bray curtis data matrix
DIV_RelA.dist <- phyloseq::distance(DIV_RelA, method = "bray")
  samp.DIV_RelA.dist <- as(sample_data(DIV_RelA), "data.frame") #sample df

#adonis by colony level
adonis2(DIV_RelA.dist ~ TagID, data = samp.DIV_RelA.dist)
    

# Homogeneity of dispersion test
betaTagID <- betadisper(DIV_RelA.dist, samp.DIV_RelA.dist$TagID)
perm<-permutest(betaTagID, pairwise = T) 
(betaTagID.HSD <- TukeyHSD(betaTagID)) #do you need? impossible to visualize
plot(betaTagID.HSD)

#adonis by sample location and colony ID
adonis2(DIV_RelA.dist ~ Location*TagID, data = samp.DIV_RelA.dist)
   
 # Homogeneity of dispersion test, Location
betaTagID <- betadisper(DIV_RelA.dist, samp.DIV_RelA.dist$Location)
perm<-permutest(betaTagID, pairwise = T) 

#adonis by broad location and colony ID
adonis2(DIV_RelA.dist ~ SampLoc*TagID, data = samp.DIV_RelA.dist)

#Ordination NMDS for all colonies
set.seed(30)
ord.DIV_RelA<- ordinate(DIV_RelA, "NMDS", "bray", trymax = 100) 
stressplot(ord.DIV_RelA)
scores.ord.DIV_RelA_stats.T0 <- as.data.frame(cbind(vegan::scores(ord.DIV_RelA, display="sites"))) #make df
scores.ord.DIV_RelA_stats.T0$TagID <- samp.DIV_RelA.dist$TagID
scores.ord.DIV_RelA_stats.T0$Samp.Date <- samp.DIV_RelA.dist$Samp.Date
scores.ord.DIV_RelA_stats.T0$Location <- samp.DIV_RelA.dist$Location

ggplot(scores.ord.DIV_RelA_stats.T0, aes(x = NMDS1, y = NMDS2)) + 
  geom_point(aes(colour = TagID), size = 4,show.legend=F) +
  #scale_color_manual(values=Sp.colors)+
  # geom_text(label = rownames(Clade), nudge_x = 0.05, nudge_y = 0.05,check_overlap = T) +
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = TagID), linetype = 2,show.legend=F) +
  ggtitle("NMDS its2 Bray Curtis TagID") +
  theme_classic()
#ggsave("bc_all.pdf")

ggplot(scores.ord.DIV_RelA_stats.T0, aes(x = NMDS1, y = NMDS2)) + 
  geom_point(aes(colour = Location), size = 4, show.legend=T) +
  #scale_color_manual(values=Sp.colors)+
  #geom_text(label = rownames(scores), nudge_x = 0.05, nudge_y = 0.05,check_overlap = T) +
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Location), linetype = 2,show.legend=F) +
  ggtitle("NMDS its2 Bray Curtis Location") +
  theme_classic()
#ggsave("bc_all.pdf")


```

Adonis within colony:
Subset by colony DIV, data setup
```{r} 
sample_variables(DIV_RelA) #what are the variables

#970
Mcap_970 <- subset_samples(DIV_RelA, TagID =="970") #Subset by colony 
  Mcap_970.dist <- phyloseq::distance(Mcap_970, method = "bray")
  samp.Mcap_970<- as(sample_data(Mcap_970), "data.frame") #sample df
  
    #adonis by colony level
    adonis2(Mcap_970.dist ~ Location, data = samp.Mcap_970)
    
#965
Mcap_965 <- subset_samples(DIV_RelA, TagID =="965") #Subset by colony  
  Mcap_965.dist <- phyloseq::distance(Mcap_965, method = "bray")
  samp.Mcap_965<- as(sample_data(Mcap_965), "data.frame") #sample df
  
    #adonis by colony level
    adonis2(Mcap_965.dist ~ Location, data = samp.Mcap_965)

#969
Mcap_969 <- subset_samples(DIV_RelA, TagID =="969") #Subset by colony  
  Mcap_969.dist <- phyloseq::distance(Mcap_969, method = "bray")
  samp.Mcap_969<- as(sample_data(Mcap_969), "data.frame") #sample df
  
    #adonis by colony level
    adonis2(Mcap_969.dist ~ Location, data = samp.Mcap_969)
    
#972
Mcap_972 <- subset_samples(DIV_RelA, TagID =="972") #Subset by colony  
  Mcap_972.dist <- phyloseq::distance(Mcap_972, method = "bray")
  samp.Mcap_972<- as(sample_data(Mcap_972), "data.frame") #sample df
  
    #adonis by colony level
    adonis2(Mcap_972.dist ~ Location, data = samp.Mcap_972)
   
#971
Mcap_971 <- subset_samples(DIV_RelA, TagID =="971") #Subset by colony  
  Mcap_971.dist <- phyloseq::distance(Mcap_971, method = "bray")
  samp.Mcap_971<- as(sample_data(Mcap_971), "data.frame") #sample df
  
    #adonis by colony level
    adonis2(Mcap_971.dist ~ Location, data = samp.Mcap_971)    

#968
Mcap_968 <- subset_samples(DIV_RelA, TagID =="968") #Subset by colony  
  Mcap_968.dist <- phyloseq::distance(Mcap_968, method = "bray")
  samp.Mcap_968<- as(sample_data(Mcap_968), "data.frame") #sample df
  
    #adonis by colony level
    adonis2(Mcap_968.dist ~ Location, data = samp.Mcap_968)    
    
    
#966
Mcap_966 <- subset_samples(DIV_RelA, TagID =="966") #Subset by colony  
  Mcap_966.dist <- phyloseq::distance(Mcap_966, method = "bray")
  samp.Mcap_966<- as(sample_data(Mcap_966), "data.frame") #sample df
  
    #adonis by colony level
    adonis2(Mcap_966.dist ~ Location, data = samp.Mcap_966)
    
#978
Mcap_978 <- subset_samples(DIV_RelA, TagID =="978") #Subset by colony  
  Mcap_978.dist <- phyloseq::distance(Mcap_978, method = "bray")
  samp.Mcap_978<- as(sample_data(Mcap_978), "data.frame") #sample df
  
    #adonis by colony level
    adonis2(Mcap_978.dist ~ Location, data = samp.Mcap_978)
    
 #963
Mcap_963 <- subset_samples(DIV_RelA, TagID =="963") #Subset by colony  
  Mcap_963.dist <- phyloseq::distance(Mcap_963, method = "bray")
  samp.Mcap_963<- as(sample_data(Mcap_963), "data.frame") #sample df
  
    #adonis by colony level
    adonis2(Mcap_963.dist ~ Location, data = samp.Mcap_963)

#961
Mcap_961 <- subset_samples(DIV_RelA, TagID =="961") #Subset by colony  
  Mcap_961.dist <- phyloseq::distance(Mcap_961, method = "bray")
  samp.Mcap_961<- as(sample_data(Mcap_961), "data.frame") #sample df
  
    #adonis by colony level
    adonis2(Mcap_961.dist ~ Location, data = samp.Mcap_961)
    
#981
Mcap_981 <- subset_samples(DIV_RelA, TagID =="981") #Subset by colony  
  Mcap_981.dist <- phyloseq::distance(Mcap_981, method = "bray")
  samp.Mcap_981<- as(sample_data(Mcap_981), "data.frame") #sample df
  
    #adonis by colony level
    adonis2(Mcap_981.dist ~ Location, data = samp.Mcap_981)
    
#967
Mcap_967 <- subset_samples(DIV_RelA, TagID =="967") #Subset by colony  
  Mcap_967.dist <- phyloseq::distance(Mcap_967, method = "bray")
  samp.Mcap_967<- as(sample_data(Mcap_967), "data.frame") #sample df
  
    #adonis by colony level
    adonis2(Mcap_967.dist ~ Location, data = samp.Mcap_967)
    
#983
Mcap_983 <- subset_samples(DIV_RelA, TagID =="983") #Subset by colony  
  Mcap_983.dist <- phyloseq::distance(Mcap_983, method = "bray")
  samp.Mcap_983<- as(sample_data(Mcap_983), "data.frame") #sample df
  
    #adonis by colony level
    adonis2(Mcap_983.dist ~ Location, data = samp.Mcap_983)
    
#974
Mcap_974 <- subset_samples(DIV_RelA, TagID =="974") #Subset by colony  
  Mcap_974.dist <- phyloseq::distance(Mcap_974, method = "bray")
  samp.Mcap_974<- as(sample_data(Mcap_974), "data.frame") #sample df
  
    #adonis by colony level
    adonis2(Mcap_974.dist ~ Location, data = samp.Mcap_974)
    
#984
Mcap_984 <- subset_samples(DIV_RelA, TagID =="984") #Subset by colony  
  Mcap_984.dist <- phyloseq::distance(Mcap_984, method = "bray")
  samp.Mcap_984<- as(sample_data(Mcap_984), "data.frame") #sample df
  
    #adonis by colony level
    adonis2(Mcap_984.dist ~ Location, data = samp.Mcap_984)
    
#962
Mcap_962 <- subset_samples(DIV_RelA, TagID =="962") #Subset by colony  
  Mcap_962.dist <- phyloseq::distance(Mcap_962, method = "bray")
  samp.Mcap_962<- as(sample_data(Mcap_962), "data.frame") #sample df
  
    #adonis by colony level
    adonis2(Mcap_962.dist ~ Location, data = samp.Mcap_962)
    
#957
Mcap_957 <- subset_samples(DIV_RelA, TagID =="957") #Subset by colony  
  Mcap_957.dist <- phyloseq::distance(Mcap_957, method = "bray")
  samp.Mcap_957<- as(sample_data(Mcap_957), "data.frame") #sample df
  
    #adonis by colony level
    adonis2(Mcap_957.dist ~ Location, data = samp.Mcap_957)
    
#945
Mcap_945 <- subset_samples(DIV_RelA, TagID =="945") #Subset by colony  
  Mcap_945.dist <- phyloseq::distance(Mcap_945, method = "bray")
  samp.Mcap_945<- as(sample_data(Mcap_945), "data.frame") #sample df
  
    #adonis by colony level
    adonis2(Mcap_945.dist ~ Location, data = samp.Mcap_945)
        
#960
Mcap_960 <- subset_samples(DIV_RelA, TagID =="960") #Subset by colony  
  Mcap_960.dist <- phyloseq::distance(Mcap_960, method = "bray")
  samp.Mcap_960<- as(sample_data(Mcap_960), "data.frame") #sample df
  
    #adonis by colony level
    adonis2(Mcap_960.dist ~ Location, data = samp.Mcap_960)
    
#976
Mcap_976 <- subset_samples(DIV_RelA, TagID =="976") #Subset by colony  
  Mcap_976.dist <- phyloseq::distance(Mcap_976, method = "bray")
  samp.Mcap_976<- as(sample_data(Mcap_976), "data.frame") #sample df
  
    #adonis by colony level
    adonis2(Mcap_976.dist ~ Location, data = samp.Mcap_976)
    
#985
Mcap_985 <- subset_samples(DIV_RelA, TagID =="985") #Subset by colony  
  Mcap_985.dist <- phyloseq::distance(Mcap_985, method = "bray")
  samp.Mcap_985<- as(sample_data(Mcap_985), "data.frame") #sample df
  
    #adonis by colony level
    adonis2(Mcap_985.dist ~ Location, data = samp.Mcap_985)
    
#964
Mcap_964 <- subset_samples(DIV_RelA, TagID =="964") #Subset by colony  
  Mcap_964.dist <- phyloseq::distance(Mcap_964, method = "bray")
  samp.Mcap_964<- as(sample_data(Mcap_964), "data.frame") #sample df
  
    #adonis by colony level
    adonis2(Mcap_964.dist ~ Location, data = samp.Mcap_964)
    
#959
Mcap_959 <- subset_samples(DIV_RelA, TagID =="959") #Subset by colony  
  Mcap_959.dist <- phyloseq::distance(Mcap_959, method = "bray")
  samp.Mcap_959<- as(sample_data(Mcap_959), "data.frame") #sample df
  
    #adonis by colony level
    adonis2(Mcap_959.dist ~ Location, data = samp.Mcap_959)
    
#980
Mcap_980 <- subset_samples(DIV_RelA, TagID =="980") #Subset by colony  
  Mcap_980.dist <- phyloseq::distance(Mcap_980, method = "bray")
  samp.Mcap_980<- as(sample_data(Mcap_980), "data.frame") #sample df
  
    #adonis by colony level
    adonis2(Mcap_980.dist ~ Location, data = samp.Mcap_980)
  
```
Plots DIVS by colony
```{r}  

#try to transform samp counts

#for the legend
 plot_bar(Mcap_970,fill = "DIV")+
      geom_bar(aes(color=DIV, fill=DIV), stat="identity", position="stack")

Mcap_970_plot<-plot_bar(Mcap_970, "Sample_name", fill="DIV", title="Mcap_970")+
   theme(legend.position="none");Mcap_970_plot
Mcap_965_plot<-plot_bar(Mcap_965, "Sample_name", fill="DIV", title="Mcap_965")+
   theme(legend.position="none");Mcap_965_plot
Mcap_969_plot<-plot_bar(Mcap_969, "Sample_name", fill="DIV", title="Mcap_969")+
   theme(legend.position="none");Mcap_969_plot
Mcap_972_plot<-plot_bar(Mcap_972, "Sample_name", fill="DIV", title="Mcap_972")+
   theme(legend.position="none");Mcap_972_plot
Mcap_971_plot<-plot_bar(Mcap_971, "Sample_name", fill="DIV", title="Mcap_971")+
   theme(legend.position="none");Mcap_971_plot
Mcap_968_plot<-plot_bar(Mcap_968, "Sample_name", fill="DIV", title="Mcap_968")+
   theme(legend.position="none");Mcap_968_plot
Mcap_966_plot<-plot_bar(Mcap_966, "Sample_name", fill="DIV", title="Mcap_966")+
   theme(legend.position="none");Mcap_966_plot
Mcap_978_plot<-plot_bar(Mcap_978, "Sample_name", fill="DIV", title="Mcap_978")+
   theme(legend.position="none");Mcap_978_plot
Mcap_963_plot<-plot_bar(Mcap_963, "Sample_name", fill="DIV", title="Mcap_963")+
   theme(legend.position="none");Mcap_963_plot
Mcap_961_plot<-plot_bar(Mcap_961, "Sample_name", fill="DIV", title="Mcap_961")+
   theme(legend.position="none");Mcap_961_plot
Mcap_981_plot<-plot_bar(Mcap_981, "Sample_name", fill="DIV", title="Mcap_981")+
   theme(legend.position="none");Mcap_981_plot
Mcap_967_plot<-plot_bar(Mcap_967, "Sample_name", fill="DIV", title="Mcap_967")+
   theme(legend.position="none");Mcap_967_plot
Mcap_983_plot<-plot_bar(Mcap_983, "Sample_name", fill="DIV", title="Mcap_983")+
   theme(legend.position="none");Mcap_983_plot
Mcap_974_plot<-plot_bar(Mcap_974, "Sample_name", fill="DIV", title="Mcap_974")+
   theme(legend.position="none");Mcap_974_plot
Mcap_984_plot<-plot_bar(Mcap_984, "Sample_name", fill="DIV", title="Mcap_984")+
   theme(legend.position="none");Mcap_984_plot
Mcap_962_plot<-plot_bar(Mcap_962, "Sample_name", fill="DIV", title="Mcap_962")+
   theme(legend.position="none");Mcap_962_plot
Mcap_957_plot<-plot_bar(Mcap_957, "Sample_name", fill="DIV", title="Mcap_957")+
   theme(legend.position="none");Mcap_957_plot
Mcap_945_plot<-plot_bar(Mcap_945, "Sample_name", fill="DIV", title="Mcap_945")+
   theme(legend.position="none");Mcap_945_plot
Mcap_960_plot<-plot_bar(Mcap_960, "Sample_name", fill="DIV", title="Mcap_960")+
   theme(legend.position="none");Mcap_960_plot
Mcap_976_plot<-plot_bar(Mcap_976, "Sample_name", fill="DIV", title="Mcap_976")+
   theme(legend.position="none");Mcap_976_plot
Mcap_985_plot<-plot_bar(Mcap_985, "Sample_name", fill="DIV", title="Mcap_985")+
   theme(legend.position="none");Mcap_985_plot
Mcap_964_plot<-plot_bar(Mcap_964, "Sample_name", fill="DIV", title="Mcap_964")+
   theme(legend.position="none");Mcap_964_plot
Mcap_985_plot<-plot_bar(Mcap_985, "Sample_name", fill="DIV", title="Mcap_985")+
   theme(legend.position="none");Mcap_985_plot
Mcap_959_plot<-plot_bar(Mcap_959, "Sample_name", fill="DIV", title="Mcap_959")+
   theme(legend.position="none");Mcap_959_plot
Mcap_980_plot<-plot_bar(Mcap_980, "Sample_name", fill="DIV", title="Mcap_980")+
   theme(legend.position="none");Mcap_980_plot

#name of sample location on C/Ds
Mcap_970_plotB<-plot_bar(Mcap_970, "LocSamp", fill="DIV", title="Mcap_970")+
   theme(legend.position="none");Mcap_970_plotB


Mcap_DIV_plot1 <- ggarrange(Mcap_970_plot, Mcap_965_plot,Mcap_969_plot,Mcap_972_plot, Mcap_971_plot,Mcap_968_plot,Mcap_966_plot,Mcap_978_plot,
                    #labels = c("A", "B", "C","D","E","F"),
                    ncol = 4, nrow = 2);Mcap_DIV_plot1

Mcap_DIV_plot2 <- ggarrange(Mcap_963_plot, Mcap_961_plot,Mcap_981_plot,Mcap_967_plot,
Mcap_983_plot,Mcap_974_plot,Mcap_984_plot,Mcap_962_plot,
                    #labels = c("A", "B", "C","D","E","F"),
                    ncol = 4, nrow = 2);Mcap_DIV_plot2

Mcap_DIV_plot3 <- ggarrange(Mcap_957_plot,Mcap_945_plot,Mcap_960_plot,Mcap_976_plot,
Mcap_985_plot,Mcap_964_plot,Mcap_985_plot,Mcap_964_plot,Mcap_985_plot,Mcap_959_plot,Mcap_980_plot,
                    #labels = c("A", "B", "C","D","E","F"),
                    ncol = 4, nrow = 3);Mcap_DIV_plot3

```

PROFILES
Subset by colony PROFILE, data setup
coral_Absolute_Pro.rare
```{r}  
 
library("RColorBrewer")
getPalette = colorRampPalette(brewer.pal(8, "Dark2"))

speciesList = unique(tax_table(coral_Absolute_Pro.rare)[,"its2_type_profile"])
speciesPalette = getPalette(length(speciesList))
names(speciesPalette) = speciesList

#scale_color_manual(values=speciesPalette)
#scale_fill_manual(values= speciesPalette)

#this works but you need better pallet color = grDevices::colors()[grep('gr(a|e)y', grDevices::colors(), invert = T)]





sample_variables(coral_Absolute_Pro.rare) #what are the variables

Mcap_970_Pro <- subset_samples(coral_Absolute_Pro.rare, TagID =="970") #Subset by colony  
Mcap_965_Pro <- subset_samples(coral_Absolute_Pro.rare, TagID =="965") #Subset by colony  
Mcap_969_Pro <- subset_samples(coral_Absolute_Pro.rare, TagID =="969") #Subset by colony  
Mcap_972_Pro <- subset_samples(coral_Absolute_Pro.rare, TagID =="972") #Subset by colony  
Mcap_971_Pro <- subset_samples(coral_Absolute_Pro.rare, TagID =="971") #Subset by colony 
Mcap_968_Pro <- subset_samples(coral_Absolute_Pro.rare, TagID =="968") #Subset by colony 
Mcap_966_Pro <- subset_samples(coral_Absolute_Pro.rare, TagID =="966") #Subset by colony 
Mcap_978_Pro <- subset_samples(coral_Absolute_Pro.rare, TagID =="978") #Subset by colony 
Mcap_963_Pro <- subset_samples(coral_Absolute_Pro.rare, TagID =="963") #Subset by colony 
Mcap_961_Pro <- subset_samples(coral_Absolute_Pro.rare, TagID =="961") #Subset by colony 
Mcap_981_Pro <- subset_samples(coral_Absolute_Pro.rare, TagID =="981") #Subset by colony 
Mcap_967_Pro <- subset_samples(coral_Absolute_Pro.rare, TagID =="967") #Subset by colony 
Mcap_983_Pro <- subset_samples(coral_Absolute_Pro.rare, TagID =="983") #Subset by colony 
Mcap_974_Pro <- subset_samples(coral_Absolute_Pro.rare, TagID =="974") #Subset by colony 
Mcap_984_Pro <- subset_samples(coral_Absolute_Pro.rare, TagID =="984") #Subset by colony 
Mcap_962_Pro <- subset_samples(coral_Absolute_Pro.rare, TagID =="962") #Subset by colony 
Mcap_957_Pro <- subset_samples(coral_Absolute_Pro.rare, TagID =="957") #Subset by colony 
Mcap_945_Pro <- subset_samples(coral_Absolute_Pro.rare, TagID =="945") #Subset by colony 
Mcap_960_Pro <- subset_samples(coral_Absolute_Pro.rare, TagID =="960") #Subset by colony 
Mcap_976_Pro <- subset_samples(coral_Absolute_Pro.rare, TagID =="976") #Subset by colony 
Mcap_985_Pro <- subset_samples(coral_Absolute_Pro.rare, TagID =="985") #Subset by colony 
Mcap_964_Pro <- subset_samples(coral_Absolute_Pro.rare, TagID =="964") #Subset by colony 
Mcap_959_Pro <- subset_samples(coral_Absolute_Pro.rare, TagID =="959") #Subset by colony 
Mcap_980_Pro <- subset_samples(coral_Absolute_Pro.rare, TagID =="980") #Subset by colony 
  
```
Plots PRO
```{r} 
#for the legend
 plot_bar(Mcap_970_Pro,fill = "its2_type_profile")+
      geom_bar(aes(color=its2_type_profile, fill=its2_type_profile), stat="identity", position="stack")


Mcap_970_PRO_plot<-plot_bar(Mcap_970_Pro, "Sample_name", fill="its2_type_profile", title="Mcap_970_Pro")+
   theme(legend.position="none");Mcap_970_PRO_plot
Mcap_965_PRO_plot<-plot_bar(Mcap_965_Pro, "Sample_name", fill="its2_type_profile", title="Mcap_965_Pro")+
   theme(legend.position="none");Mcap_965_PRO_plot
Mcap_969_PRO_plot<-plot_bar(Mcap_969_Pro, "Sample_name", fill="its2_type_profile", title="Mcap_969_Pro")+
   theme(legend.position="none");Mcap_969_PRO_plot
Mcap_972_PRO_plot<-plot_bar(Mcap_972_Pro, "Sample_name", fill="its2_type_profile", title="Mcap_972_Pro")+
   theme(legend.position="none");Mcap_972_PRO_plot
Mcap_971_PRO_plot<-plot_bar(Mcap_971_Pro, "Sample_name", fill="its2_type_profile", title="Mcap_971_Pro")+
   theme(legend.position="none");Mcap_971_PRO_plot
Mcap_968_PRO_plot<-plot_bar(Mcap_968_Pro, "Sample_name", fill="its2_type_profile", title="Mcap_968_Pro")+
   theme(legend.position="none");Mcap_968_PRO_plot
Mcap_966_PRO_plot<-plot_bar(Mcap_966_Pro, "Sample_name", fill="its2_type_profile", title="Mcap_966_Pro")+
   theme(legend.position="none");Mcap_966_PRO_plot
Mcap_978_PRO_plot<-plot_bar(Mcap_978_Pro, "Sample_name", fill="its2_type_profile", title="Mcap_978_Pro")+
   theme(legend.position="none");Mcap_978_PRO_plot
Mcap_963_PRO_plot<-plot_bar(Mcap_963_Pro, "Sample_name", fill="its2_type_profile", title="Mcap_963_Pro")+
   theme(legend.position="none");Mcap_963_PRO_plot
Mcap_961_PRO_plot<-plot_bar(Mcap_961_Pro, "Sample_name", fill="its2_type_profile", title="Mcap_961_Pro")+
   theme(legend.position="none");Mcap_961_PRO_plot
Mcap_981_PRO_plot<-plot_bar(Mcap_981_Pro, "Sample_name", fill="its2_type_profile", title="Mcap_981_Pro")+
   theme(legend.position="none");Mcap_981_PRO_plot
Mcap_967_PRO_plot<-plot_bar(Mcap_967_Pro, "Sample_name", fill="its2_type_profile", title="Mcap_967_Pro")+
   theme(legend.position="none");Mcap_967_PRO_plot
Mcap_983_PRO_plot<-plot_bar(Mcap_983_Pro, "Sample_name", fill="its2_type_profile", title="Mcap_983_Pro")+
   theme(legend.position="none");Mcap_983_PRO_plot
Mcap_974_PRO_plot<-plot_bar(Mcap_974_Pro, "Sample_name", fill="its2_type_profile", title="Mcap_974_Pro")+
   theme(legend.position="none");Mcap_974_PRO_plot
Mcap_984_PRO_plot<-plot_bar(Mcap_984_Pro, "Sample_name", fill="its2_type_profile", title="Mcap_984_Pro")+
   theme(legend.position="none");Mcap_984_PRO_plot
Mcap_962_PRO_plot<-plot_bar(Mcap_962_Pro, "Sample_name", fill="its2_type_profile", title="Mcap_962_Pro")+
   theme(legend.position="none");Mcap_962_PRO_plot
Mcap_957_PRO_plot<-plot_bar(Mcap_957_Pro, "Sample_name", fill="its2_type_profile", title="Mcap_957_Pro")+
   theme(legend.position="none");Mcap_957_PRO_plot
Mcap_945_PRO_plot<-plot_bar(Mcap_945_Pro, "Sample_name", fill="its2_type_profile", title="Mcap_945_Pro")+
   theme(legend.position="none");Mcap_945_PRO_plot
Mcap_960_PRO_plot<-plot_bar(Mcap_960_Pro, "Sample_name", fill="its2_type_profile", title="Mcap_960_Pro")+
   theme(legend.position="none");Mcap_960_PRO_plot
Mcap_976_PRO_plot<-plot_bar(Mcap_976_Pro, "Sample_name", fill="its2_type_profile", title="Mcap_976_Pro")+
   theme(legend.position="none");Mcap_976_PRO_plot
Mcap_985_PRO_plot<-plot_bar(Mcap_985_Pro, "Sample_name", fill="its2_type_profile", title="Mcap_985_Pro")+
   theme(legend.position="none");Mcap_985_PRO_plot
Mcap_964_PRO_plot<-plot_bar(Mcap_964_Pro, "Sample_name", fill="its2_type_profile", title="Mcap_964_Pro")+
   theme(legend.position="none");Mcap_964_PRO_plot
Mcap_985_PRO_plot<-plot_bar(Mcap_985_Pro, "Sample_name", fill="its2_type_profile", title="Mcap_985_Pro")+
   theme(legend.position="none");Mcap_985_PRO_plot
Mcap_959_PRO_plot<-plot_bar(Mcap_959_Pro, "Sample_name", fill="its2_type_profile", title="Mcap_959_Pro")+
   theme(legend.position="none");Mcap_959_PRO_plot
Mcap_980_PRO_plot<-plot_bar(Mcap_980_Pro, "Sample_name", fill="its2_type_profile", title="Mcap_980_Pro")+
   theme(legend.position="none");Mcap_980_PRO_plot


Mcap_DIV_PRO_plot1 <- ggarrange(Mcap_970_PRO_plot, Mcap_965_PRO_plot,Mcap_969_PRO_plot,Mcap_972_PRO_plot, Mcap_971_PRO_plot,Mcap_968_PRO_plot,Mcap_966_PRO_plot,Mcap_978_PRO_plot,
                    #labels = c("A", "B", "C","D","E","F"),
                    ncol = 4, nrow = 2);Mcap_DIV_PRO_plot1

Mcap_DIV_PRO_plot2 <- ggarrange(Mcap_963_PRO_plot, Mcap_961_PRO_plot,Mcap_981_PRO_plot,Mcap_967_PRO_plot,
Mcap_983_PRO_plot,Mcap_974_PRO_plot,Mcap_984_PRO_plot,Mcap_962_PRO_plot,
                    #labels = c("A", "B", "C","D","E","F"),
                    ncol = 4, nrow = 2);Mcap_DIV_PRO_plot2

Mcap_DIV_PRO_plot3 <- ggarrange(Mcap_957_PRO_plot,Mcap_945_PRO_plot,Mcap_960_PRO_plot,Mcap_976_PRO_plot,
Mcap_985_PRO_plot,Mcap_964_PRO_plot,Mcap_985_PRO_plot,Mcap_964_PRO_plot,Mcap_985_PRO_plot,Mcap_959_PRO_plot,Mcap_980_PRO_plot,
                    #labels = c("A", "B", "C","D","E","F"),
                    ncol = 4, nrow = 3);Mcap_DIV_PRO_plot3

```
Plots CLADE
```{r}   

scale_color_manual(values=speciesPalette)
scale_fill_manual(values= speciesPalette)


 Mcap_970_clade_plot<-plot_bar(Mcap_970_Pro, "LocSamp", fill="clade", title="Mcap_970_clade");Mcap_970_clade_plot


# pretty the axis labels with ggplot2
library("ggplot2")
Mcap_970_clade_plot + theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust=1))


Mcap_970_clade_plot<-plot_bar(Mcap_970_Pro, "LocSamp",fill="clade", title="Mcap_970_clade")+
   theme(legend.position="none") + 
  theme(axis.text.x = element_text(angle = 90))+
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(), axis.line = element_line(colour = "black"));Mcap_970_clade_plot

Mcap_965_clade_plot<-plot_bar(Mcap_965_Pro, "LocSamp", fill="clade", title="Mcap_965_clade")+
   theme(legend.position="none") +  theme(axis.text.x = element_text(angle = 90))+
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(), axis.line = element_line(colour = "black"));Mcap_965_clade_plot

Mcap_969_clade_plot<-plot_bar(Mcap_969_Pro, "LocSamp", fill="clade", title="Mcap_969_clade")+
   theme(legend.position="none") +  theme(axis.text.x = element_text(angle = 90))+
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(), axis.line = element_line(colour = "black"));Mcap_969_clade_plot

Mcap_972_clade_plot<-plot_bar(Mcap_972_Pro, "LocSamp", fill="clade", title="Mcap_972_clade")+
   theme(legend.position="none") +  theme(axis.text.x = element_text(angle = 90))+
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(), axis.line = element_line(colour = "black"));Mcap_972_clade_plot

Mcap_971_clade_plot<-plot_bar(Mcap_971_Pro, "LocSamp", fill="clade", title="Mcap_971_clade")+
   theme(legend.position="none") +  theme(axis.text.x = element_text(angle = 90))+
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(), axis.line = element_line(colour = "black"));Mcap_971_clade_plot

Mcap_968_clade_plot<-plot_bar(Mcap_968_Pro, "LocSamp", fill="clade", title="Mcap_968_clade")+
   theme(legend.position="none") +  theme(axis.text.x = element_text(angle = 90))+
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(), axis.line = element_line(colour = "black"));Mcap_968_clade_plot

Mcap_966_clade_plot<-plot_bar(Mcap_966_Pro, "LocSamp", fill="clade", title="Mcap_966_clade")+
   theme(legend.position="none") +  theme(axis.text.x = element_text(angle = 90))+
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(), axis.line = element_line(colour = "black"));Mcap_966_clade_plot

Mcap_978_clade_plot<-plot_bar(Mcap_978_Pro, "LocSamp", fill="clade", title="Mcap_978_clade")+
   theme(legend.position="none") +  theme(axis.text.x = element_text(angle = 90))+
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(), axis.line = element_line(colour = "black"));Mcap_978_clade_plot

Mcap_963_clade_plot<-plot_bar(Mcap_963_Pro, "LocSamp", fill="clade", title="Mcap_963_clade")+
   theme(legend.position="none") +  theme(axis.text.x = element_text(angle = 90))+
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(), axis.line = element_line(colour = "black"));Mcap_963_clade_plot

Mcap_961_clade_plot<-plot_bar(Mcap_961_Pro, "LocSamp", fill="clade", title="Mcap_961_clade")+
   theme(legend.position="none") +  theme(axis.text.x = element_text(angle = 90))+
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(), axis.line = element_line(colour = "black"));Mcap_961_clade_plot

Mcap_981_clade_plot<-plot_bar(Mcap_981_Pro, "LocSamp", fill="clade", title="Mcap_981_clade")+
   theme(legend.position="none") +  theme(axis.text.x = element_text(angle = 90))+
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(), axis.line = element_line(colour = "black"));Mcap_981_clade_plot

Mcap_967_clade_plot<-plot_bar(Mcap_967_Pro, "LocSamp", fill="clade", title="Mcap_967_clade")+
   theme(legend.position="none") +  theme(axis.text.x = element_text(angle = 90))+
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(), axis.line = element_line(colour = "black"));Mcap_967_clade_plot

Mcap_983_clade_plot<-plot_bar(Mcap_983_Pro, "LocSamp", fill="clade", title="Mcap_983_clade")+
   theme(legend.position="none") +  theme(axis.text.x = element_text(angle = 90))+
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(), axis.line = element_line(colour = "black"));Mcap_983_clade_plot

Mcap_974_clade_plot<-plot_bar(Mcap_974_Pro, "LocSamp", fill="clade", title="Mcap_974_clade")+
   theme(legend.position="none") +  theme(axis.text.x = element_text(angle = 90))+
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(), axis.line = element_line(colour = "black"));Mcap_974_clade_plot

Mcap_984_clade_plot<-plot_bar(Mcap_984_Pro, "LocSamp", fill="clade", title="Mcap_984_clade")+
   theme(legend.position="none") +  theme(axis.text.x = element_text(angle = 90))+
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(), axis.line = element_line(colour = "black"));Mcap_984_clade_plot

Mcap_962_clade_plot<-plot_bar(Mcap_962_Pro, "LocSamp", fill="clade", title="Mcap_962_clade")+
   theme(legend.position="none") +  theme(axis.text.x = element_text(angle = 90))+
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(), axis.line = element_line(colour = "black"));Mcap_962_clade_plot

Mcap_957_clade_plot<-plot_bar(Mcap_957_Pro, "LocSamp", fill="clade", title="Mcap_957_clade")+
   theme(legend.position="none") +  theme(axis.text.x = element_text(angle = 90))+
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(), axis.line = element_line(colour = "black"));Mcap_957_clade_plot

Mcap_945_clade_plot<-plot_bar(Mcap_945_Pro, "LocSamp", fill="clade", title="Mcap_945_clade")+
   theme(legend.position="none") +  theme(axis.text.x = element_text(angle = 90))+
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(), axis.line = element_line(colour = "black"));Mcap_945_clade_plot

Mcap_960_clade_plot<-plot_bar(Mcap_960_Pro, "LocSamp", fill="clade", title="Mcap_960_clade")+
   theme(legend.position="none") +  theme(axis.text.x = element_text(angle = 90))+
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(), axis.line = element_line(colour = "black"));Mcap_960_clade_plot

Mcap_976_clade_plot<-plot_bar(Mcap_976_Pro, "LocSamp", fill="clade", title="Mcap_976_clade")+
   theme(legend.position="none") +  theme(axis.text.x = element_text(angle = 90))+
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(), axis.line = element_line(colour = "black"));Mcap_976_clade_plot

Mcap_985_clade_plot<-plot_bar(Mcap_985_Pro, "LocSamp", fill="clade", title="Mcap_985_clade")+
   theme(legend.position="none") +  theme(axis.text.x = element_text(angle = 90))+
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(), axis.line = element_line(colour = "black"));Mcap_985_clade_plot

Mcap_964_clade_plot<-plot_bar(Mcap_964_Pro, "LocSamp", fill="clade", title="Mcap_964_clade")+
   theme(legend.position="none") +  theme(axis.text.x = element_text(angle = 90))+
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(), axis.line = element_line(colour = "black"));Mcap_964_clade_plot

Mcap_985_clade_plot<-plot_bar(Mcap_985_Pro, "LocSamp", fill="clade", title="Mcap_985_clade")+
   theme(legend.position="none") +  theme(axis.text.x = element_text(angle = 90))+
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(), axis.line = element_line(colour = "black"));Mcap_985_clade_plot

Mcap_959_clade_plot<-plot_bar(Mcap_959_Pro, "LocSamp", fill="clade", title="Mcap_959_clade")+
   theme(legend.position="none") +  theme(axis.text.x = element_text(angle = 90))+
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(), axis.line = element_line(colour = "black"));Mcap_959_clade_plot

Mcap_980_clade_plot<-plot_bar(Mcap_980_Pro, "LocSamp", fill="clade", title="Mcap_980_clade")+
   theme(legend.position="none")+theme(axis.text.x = element_text(angle = 90))+
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(), axis.line = element_line(colour = "black"));Mcap_980_clade_plot


Mcap_DIV_clade_plot1 <- ggarrange(Mcap_970_clade_plot, Mcap_965_clade_plot,Mcap_969_clade_plot,Mcap_972_clade_plot, Mcap_971_clade_plot,Mcap_968_clade_plot,Mcap_966_clade_plot,Mcap_978_clade_plot,
                    #labels = c("A", "B", "C","D","E","F"),
                    ncol = 4, nrow = 2);Mcap_DIV_clade_plot1

Mcap_DIV_clade_plot2 <- ggarrange(Mcap_963_clade_plot, Mcap_961_clade_plot,Mcap_981_clade_plot,Mcap_967_clade_plot,
Mcap_983_clade_plot,Mcap_974_clade_plot,Mcap_984_clade_plot,Mcap_962_clade_plot,
                    #labels = c("A", "B", "C","D","E","F"),
                    ncol = 4, nrow = 2);Mcap_DIV_clade_plot2

Mcap_DIV_clade_plot3 <- ggarrange(Mcap_957_clade_plot,Mcap_945_clade_plot,Mcap_960_clade_plot,Mcap_976_clade_plot,
Mcap_985_clade_plot,Mcap_964_clade_plot,Mcap_985_clade_plot,Mcap_964_clade_plot,Mcap_985_clade_plot,Mcap_959_clade_plot,Mcap_980_clade_plot,
                    #labels = c("A", "B", "C","D","E","F"),
                    ncol = 4, nrow = 3);Mcap_DIV_clade_plot3
```

#get %C and %D for each sample (telomere data as well)
```{r}
#do tax glom in the data to group C and D 
 data.clade<- tax_glom(coral2.prune, taxrank="clade") 

#coerce into df
  # Extract abundance matrix from the phyloseq object
  OTU1 = as(otu_table(data.clade), "matrix")
  # transpose if necessary
  if(taxa_are_rows(data.clade)){OTU1 <- t(OTU1)}
  # Coerce to data.frame
  OTUdf = as.data.frame(OTU1)

  library(tidyverse)
library(tibble)
  
OTUdf <- tibble::rownames_to_column(OTUdf, "sample.name")
as_tibble(OTUdf)

OTUdf<-OTUdf%>%
  rename(C1 = `87732`) %>%
  rename( C2 = `87257`) %>%
  rename( D1 = `87918`) %>%
  rename( D2 = `87537`)


OTUdf<-
  OTUdf %>% rowwise() %>%
  mutate(C = sum(c(C1+C2))) %>%  #sum C
    mutate(D = sum(c(D1+D2)))   #sum D
        #select( c(1, 6:7))              # remove cols you don't want anymore, not working

OTUdf<-OTUdf %>%
    mutate(samp.num = as.numeric(str_extract(sample.name, "[0-9]+"))) %>%   #extract number in samp name
      mutate(samp.let = as.character(str_extract(sample.name, "[A-Z]+")))   # extract letter in samp name
#Next enter genotype letter combo

tibble(OTUdf) 

OTUdf<-OTUdf%>%              
  rowwise() %>%
      mutate(
        C = sum(c_across(2:3)),
        D = sum(c_across(4:5)))

write.csv(OTUdf, "Clade.samps2022.csv")     
  


```


Plot all clade level by colony
```{r} 
theme_set(theme_classic())
library(ggpubr)

library(ggarrange)


figureMCL1 <- ggarrange(Mcap_970_clade_plot, Mcap_965_clade_plot, Mcap_969_clade_plot,Mcap_972_clade_plot, 
                               Mcap_970_PRO_plot, Mcap_965_PRO_plot, Mcap_969_PRO_plot, Mcap_972_PRO_plot, 
                               Mcap_970_plot, Mcap_965_plot, Mcap_969_plot, Mcap_972_plot, 
                    ncol = 4, nrow = 3);figureMCL1

figureMCL2 <- ggarrange(Mcap_971_clade_plot,Mcap_968_clade_plot,Mcap_966_clade_plot,Mcap_978_clade_plot, 
                        Mcap_971_PRO_plot,Mcap_968_PRO_plot,Mcap_966_PRO_plot,Mcap_978_PRO_plot, 
                        Mcap_971_plot,Mcap_968_plot,Mcap_966_plot,Mcap_978_plot, 
                    ncol = 4, nrow = 3);figureMCL2

figureMCL3 <- ggarrange(Mcap_963_clade_plot, Mcap_961_clade_plot,Mcap_981_clade_plot,Mcap_967_clade_plot,
                        Mcap_963_PRO_plot, Mcap_961_PRO_plot,Mcap_981_PRO_plot,Mcap_967_PRO_plot,
                        Mcap_963_plot, Mcap_961_plot,Mcap_981_plot,Mcap_967_plot,
                    ncol = 4, nrow = 3);figureMCL3

figureMCL4 <- ggarrange(Mcap_983_clade_plot, Mcap_974_clade_plot,Mcap_984_clade_plot,Mcap_962_clade_plot, 
                        Mcap_983_PRO_plot, Mcap_974_PRO_plot,Mcap_984_PRO_plot,Mcap_962_PRO_plot, 
                        Mcap_983_plot,Mcap_974_plot,Mcap_984_plot,Mcap_962_plot, 
                    ncol = 4, nrow = 3);figureMCL4

figureMCL5 <- ggarrange(Mcap_957_clade_plot, Mcap_945_clade_plot,Mcap_960_clade_plot,Mcap_976_clade_plot,
                        Mcap_957_PRO_plot,Mcap_945_PRO_plot,Mcap_960_PRO_plot,Mcap_976_PRO_plot,
                        Mcap_957_plot,Mcap_945_plot,Mcap_960_plot,Mcap_976_plot,
                    ncol = 4, nrow = 3);figureMCL5


figureMCL6 <- ggarrange(Mcap_985_clade_plot, Mcap_964_clade_plot, Mcap_959_clade_plot,Mcap_980_clade_plot,
                        Mcap_985_PRO_plot, Mcap_964_PRO_plot, Mcap_959_PRO_plot,Mcap_980_PRO_plot,
                        Mcap_985_plot,Mcap_964_plot,Mcap_959_plot,Mcap_980_plot,
                    ncol = 4, nrow = 3);figureMCL6



```

Ordination OLD
```{r}
plot_ordination(coralDIV2.prune, ordinate(coralDIV2.prune, "MDS"), color = "TagID") + geom_point(size = 5)
plot_heatmap(coralDIV2.prune, "NMDS", "bray", "TagID", "DIV")

GP.ord <- ordinate(coralDIV2.prune, "NMDS", "bray")
p1 = plot_ordination(coralDIV2.prune, GP.ord, type="samples", color="TagID", title="taxa")
print(p1)

p2 = plot_ordination(coralDIV2.prune, GP.ord, type="samples", color="TagID", shape="TagID") 
p2 + geom_polygon(aes(fill=SampleType)) + geom_point(size=5) + ggtitle("samples")
```



# PERMANOVA coralDIV2.prune (older)

Write specific questions to answer:
-do algal communities differe between colonies? 
-do algal communities differ within colonies
-do they differ between C-only and D-only? 
-is there a dominant C type or D type
- check hypotheses based on location (tips vs branches) across entire dataset and by dom clade
```{r}            
# use this  
otu_table(DIV_RelA)[1:5, 1:5] #check rela worked

#bray curtis presence-absense and abundance
 
  DIV_RelA = prune_taxa(taxa_sums(DIV_RelA) > 0, DIV_RelA) #this removes any OTU with 0s

#bray curtis presence-absense and abundance

#all speices
set.seed(300)

#make bray curtis data matrix
bc.all <- phyloseq::distance(DIV_RelA, method = "bray")
  samp.df <- as(sample_data(coralDIV2.prune_stats), "data.frame") #sample df
  
  samp.df$TagID<-as.factor(samp.df$TagID)
    samp.df$Land_Ocean<-as.factor(samp.df$Land_Ocean)
    samp.df$SampLoc<-as.factor(samp.df$SampLoc)
    samp.df$Location<-paste(samp.df$SampLoc, samp.df$Sub.loc) #sample location and sub location
  
#adonis test
adonis2(bc.all ~ TagID*Location, data = samp.df) #all sig
    # genotype, sample location and interaction all significant. 

# Homogeneity of dispersion test
beta.colony <- betadisper(bc.all, samp.df$TagID)
perm<-permutest(beta.colony, pairwise = T) 
(bc.all.HSD <- TukeyHSD(beta.colony)) #do you need?
plot(bc.all.HSD)


#Ordination NMDS for all species
set.seed(30)
ord.DIV_bc.all <- ordinate(coralDIV2.prune_stats, "NMDS", "bray", trymax = 300)  ### NO CONVERGENCE
stressplot(ord.DIV_bc.all)
ord.DIV_bc.all <- as.data.frame(cbind(vegan::scores(ord.DIV_bc.all, display="sites"))) #make df
ord.DIV_bc.all$TagID <- samp.df$TagID
ord.DIV_bc.all$SampLoc <- samp.df$SampLoc
ord.DIV_bc.all$Zoox <- samp.df$Zoox

ggplot(ord.DIV_bc.all, aes(x = NMDS1, y = NMDS2)) + 
  geom_point(aes(colour = TagID), size = 4,show.legend=T) +
  #scale_color_manual(values=Sp.colors)+
  # geom_text(label = rownames(Clade), nudge_x = 0.05, nudge_y = 0.05,check_overlap = T) +
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = TagID), linetype = 2,show.legend=T) +
  ggtitle("NMDS its2 Bray Curtis genotype ") +
  theme_classic()
#ggsave("bc_all.pdf")

ggplot(ord.DIV_bc.all, aes(x = NMDS1, y = NMDS2, color = TagID )) +
  geom_point(aes(colour=TagID), size=4) +
  stat_ellipse(geom = "polygon",
               aes(fill = TagID), 
               alpha = 0.25)+
    theme_classic()

zoox.cols <- c("C" = "green2","CD"="darkslateblue","D"="darkorange")

ggplot(ord.DIV_bc.all, aes(x = NMDS1, y = NMDS2, color = Zoox )) +
  geom_point(aes(colour=Zoox), size=5, alpha=0.8) +
      scale_color_manual(values=zoox.cols)+
  # stat_ellipse(geom = "polygon",
  #              aes(fill = Zoox), 
  #              alpha = 0.2)+
    theme_classic()

######### Try by Cs only #### 
coralDIV2.prune_stats_C<-subset_samples(coralDIV2.prune_stats, Zoox=="C")
coralDIV2.prune_stats_C = prune_taxa(taxa_sums(coralDIV2.prune_stats_C) > 0, coralDIV2.prune_stats_C)
  
set.seed(30)
ord.DIV_bc.C <- ordinate(coralDIV2.prune_stats_C, "NMDS", "bray", trymax = 300)  ### NO CONVERGENCE
stressplot(ord.DIV_bc.C)
  samp.df.C <- as(sample_data(coralDIV2.prune_stats_C), "data.frame") #sample df

ord.DIV_bc.C <- as.data.frame(cbind(vegan::scores(ord.DIV_bc.C, display="sites"))) #make df
ord.DIV_bc.C$TagID <- samp.df.C$TagID
ord.DIV_bc.C$TagID<-as.factor(ord.DIV_bc.C$TagID)
ord.DIV_bc.C$SampLoc <- samp.df.C$SampLoc
ord.DIV_bc.C$Location<-samp.df.C$Sub.loc
ord.DIV_bc.C$Code<-paste(ord.DIV_bc.C$SampLoc, ord.DIV_bc.C$Location)
ord.DIV_bc.C$Code<-as.factor(ord.DIV_bc.C$Code)

ggplot(ord.DIV_bc.C, aes(x = NMDS1, y = NMDS2)) + 
  geom_point(aes(colour = TagID), size = 4,show.legend=T) +
  #scale_color_manual(values=Sp.colors)+
  # geom_text(label = rownames(Clade), nudge_x = 0.05, nudge_y = 0.05,check_overlap = T) +
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = TagID),show.legend=T) +
  ggtitle("NMDS its2 C only ") +
  theme_classic()
#ggsave("bc_all.pdf")

ggplot(ord.DIV_bc.C, aes(x = NMDS1, y = NMDS2, color = Code, linetype=Code)) +
  geom_point(aes(shape=Code), size=4) +
  stat_ellipse(geom = "polygon",
               aes(fill = Code), 
               alpha = 0.25)+
    theme_classic()

ggplot(ord.DIV_bc.C, aes(x = NMDS1, y = NMDS2)) + 
  geom_point(aes(colour = Code), size = 4,show.legend=T) +
  #scale_Dolor_manual(values=Sp.Dolors)+
  # geom_text(label = rownames(Clade), nudge_x = 0.05, nudge_y = 0.05,check_overlap = T) +
  # stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Code),show.legend=T) +
    geom_mark_hull(concavity = 5,expand=0,radius=0,aes(fill=Code))+
  ggtitle("NMDS its2 C only ") +
  theme_classic()
#ggsave("bc_all.pdf")



ggplot(ord.DIV_bc.C, aes(x = NMDS1, y = NMDS2)) + 
  geom_point(aes(colour = Code), size = 4,show.legend=T) +
  #scale_Dolor_manual(values=Sp.Dolors)+
  # geom_text(label = rownames(Clade), nudge_x = 0.05, nudge_y = 0.05,check_overlap = T) +
  # stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Code),show.legend=T) +
  geom_mark_ellipse(expand = 0,aes(fill=Code))+
  ggtitle("NMDS its2 C only ") +
  theme_classic()
#ggsave("bc_all.pdf")




######### Try by Ds only #### 
coralDIV2.prune_stats_D<-subset_samples(coralDIV2.prune_stats, Zoox=="D")
coralDIV2.prune_stats_D = prune_taxa(taxa_sums(coralDIV2.prune_stats_D) > 0, coralDIV2.prune_stats_D)
  
set.seed(300)
ord.DIV_bc.D <- ordinate(coralDIV2.prune_stats_D, "NMDS", "bray", trymax = 500)  #
stressplot(ord.DIV_bc.D)
  samp.df.D <- as(sample_data(coralDIV2.prune_stats_D), "data.frame") #sample df

ord.DIV_bc.D <- as.data.frame(cbind(vegan::scores(ord.DIV_bc.D, display="sites"))) #make df
ord.DIV_bc.D$TagID <- samp.df.D$TagID
ord.DIV_bc.D$TagID<-as.factor(ord.DIV_bc.D$TagID)
ord.DIV_bc.D$SampLoc <- samp.df.D$SampLoc
ord.DIV_bc.D$Location<-samp.df.D$Sub.loc
ord.DIV_bc.D$Code<-paste(ord.DIV_bc.D$SampLoc, ord.DIV_bc.D$Location)
ord.DIV_bc.D$Code<-as.factor(ord.DIV_bc.D$Code)

ggplot(ord.DIV_bc.D, aes(x = NMDS1, y = NMDS2)) + 
  geom_point(aes(colour = TagID), size = 4,show.legend=T) +
  #scale_Dolor_manual(values=Sp.Dolors)+
  # geom_text(label = rownames(Clade), nudge_x = 0.05, nudge_y = 0.05,check_overlap = T) +
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = TagID),show.legend=T) +
  ggtitle("NMDS its2 D only ") +
  theme_classic()
#ggsave("bc_all.pdf")


library(ggalt)
library(ggforce)
library(concaveman)

ggplot(ord.DIV_bc.D, aes(x = NMDS1, y = NMDS2)) + 
  geom_point(aes(colour = Code), size = 4,show.legend=T) +
  #scale_Dolor_manual(values=Sp.Dolors)+
  # geom_text(label = rownames(Clade), nudge_x = 0.05, nudge_y = 0.05,check_overlap = T) +
  # stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Code),show.legend=T) +
    geom_mark_hull(concavity = 5,expand=0,radius=0,aes(fill=Code))+
  ggtitle("NMDS its2 D only ") +
  theme_classic()
#ggsave("bc_all.pdf")


ggplot(ord.DIV_bc.D, aes(x = NMDS1, y = NMDS2, color = Code, linetype=Code)) +
  geom_point(aes(colour=Code), size=4) +
  stat_ellipse(geom = "polygon",
               aes(fill = Code), 
               alpha = 0.25)+
    theme_classic()





#bray curtis presence-absense and abundance

#all speices
set.seed(30)

#make bray curtis data matrix
bc.all <- phyloseq::distance(DIV_RelA_stats, method = "bray")
  samp.df <- as(sample_data(DIV_RelA_stats), "data.frame") #sample df
#adonis test
adonis(bc.all ~ Species*Time.Point*Treatment, data = samp.df)
    

# Homogeneity of dispersion test
betaAll <- betadisper(bc.all, samp.df$Species)
permutest(betaAll, pairwise = T) 
(bc.all.HSD <- TukeyHSD(betaAll)) #do you need?
plot(bc.all.HSD)


#Ordination NMDS for all species
set.seed(30)
ord.DIV_RelA_stats <- ordinate(DIV_RelA_stats, "NMDS", "bray", trymax = 100) 
stressplot(ord.DIV_RelA_stats)
scores.ord.DIV_RelA_stats <- as.data.frame(cbind(vegan::scores(ord.DIV_RelA_stats, display="sites"))) #make df
scores.ord.DIV_RelA_stats$Treatment <- samp.df$Treatment
scores.ord.DIV_RelA_stats$Species <- samp.df$Species
scores.ord.DIV_RelA_stats$Clade <- samp.df$Clade
scores.ord.DIV_RelA_stats$Time.Point <- samp.df$Time.Point

ggplot(scores.ord.DIV_RelA_stats, aes(x = NMDS1, y = NMDS2)) + 
  geom_point(aes(colour = Species), size = 4) +
  #scale_color_manual(values=Mcap.Parent.colors)+
  #geom_text(label = rownames(scores), nudge_x = 0.05, nudge_y = 0.05,check_overlap = T) +
  #stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment), linetype = 2) +
  ggtitle("NMDS its2 Bray Curtis") +
  theme_classic()
#ggsave("bc_all.pdf")

ggplot(scores.ord.DIV_RelA_stats, aes(x = NMDS1, y = NMDS2)) + 
  geom_point(aes(colour = Clade, shape=Species), size = 4) +
  #scale_color_manual(values=Mcap.Parent.colors)+
  #geom_text(label = rownames(scores), nudge_x = 0.05, nudge_y = 0.05,check_overlap = T) +
  #stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Treatment), linetype = 2) +
  ggtitle("NMDS its2 Bray Curtis") +
  theme_classic()
#ggsave("bc_all.pdf")



######### Try by CDs only #### 
coralDIV2.prune_stats_CD<-subset_samples(coralDIV2.prune_stats, Zoox=="CD")
coralDIV2.prune_stats_CD = prune_taxa(taxa_sums(coralDIV2.prune_stats_CD) > 0, coralDIV2.prune_stats_CD)
  
set.seed(30)
ord.DIV_bc.CD <- ordinate(coralDIV2.prune_stats_CD, "NMDS", "bray", trymax = 300)  ###  CONVERGENCE
stressplot(ord.DIV_bc.CD)
  samp.df.CD <- as(sample_data(coralDIV2.prune_stats_CD), "data.frame") #sample df

ord.DIV_bc.CD <- as.data.frame(cbind(vegan::scores(ord.DIV_bc.CD, display="sites"))) #make df
ord.DIV_bc.CD$TagID <- samp.df.CD$TagID
ord.DIV_bc.CD$TagID<-as.factor(ord.DIV_bc.CD$TagID)
ord.DIV_bc.CD$SampLoc <- samp.df.CD$SampLoc
ord.DIV_bc.CD$Location<-samp.df.CD$Sub.loc
ord.DIV_bc.CD$Code<-paste(ord.DIV_bc.CD$SampLoc, ord.DIV_bc.CD$Location)
ord.DIV_bc.CD$Code<-as.factor(ord.DIV_bc.CD$Code)

ggplot(ord.DIV_bc.CD, aes(x = NMDS1, y = NMDS2)) + 
  geom_point(aes(colour = TagID), size = 4,show.legend=T) +
  #scale_CDolor_manual(values=Sp.CDolors)+
  # geom_text(label = rownames(Clade), nudge_x = 0.05, nudge_y = 0.05,check_overlap = T) +
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = TagID),show.legend=T) +
  ggtitle("NMDS its2 CD only ") +
  theme_classic()
#ggsave("bc_all.pdf")

ggplot(ord.DIV_bc.CD, aes(x = NMDS1, y = NMDS2, color = TagID)) +
  geom_point(aes(colour=TagID), size=4, alpha=.2) +
  stat_ellipse(geom = "polygon",
               aes(fill = TagID), 
               alpha = 0.55)+
    theme_classic()

ggplot(ord.DIV_bc.CD, aes(x = NMDS1, y = NMDS2)) + 
  geom_point(aes(colour = Code), size = 4,show.legend=T) +
  #scale_Dolor_manual(values=Sp.Dolors)+
  # geom_text(label = rownames(Clade), nudge_x = 0.05, nudge_y = 0.05,check_overlap = T) +
  # stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Code),show.legend=T) +
    geom_mark_hull(concavity = 5,expand=0,radius=0,aes(fill=Code))+
  ggtitle("NMDS its2 CD only ") +
  theme_classic()
#ggsave("bc_all.pdf")



##############   BY COLONY   ##################

#945 ####
DIV_RelA_stats_945<-subset_samples(coralDIV2.prune_stats, TagID=="945")
  DIV_RelA_stats_945 = prune_taxa(taxa_sums(DIV_RelA_stats_945) > 0, DIV_RelA_stats_945) #this removes any OTU with 0s
  samp.df_945 <- as(sample_data(DIV_RelA_stats_945), "data.frame") #sample df

#all speices
set.seed(30)

#make bray curtis data matrix
bc.945 <- phyloseq::distance(coralDIV2.prune_stats, method = "bray")
  samp.df_945 <- as(sample_data(coralDIV2.prune_stats), "data.frame") #sample df
#adonis test
adonis(bc.945 ~ SampLoc, data = samp.df_945)
    


#Ordination NMDS for all species
set.seed(30)
ord.DIV_RelA_stats.945 <- ordinate(DIV_RelA_stats_945, "NMDS", "bray", trymax = 100)
  samp.df_945 <- as(sample_data(DIV_RelA_stats_945), "data.frame") #sample df
stressplot(ord.DIV_RelA_stats.945)
ord.DIV_RelA_stats.945 <- as.data.frame(cbind(vegan::scores(ord.DIV_RelA_stats.945, display="sites"))) #make df
ord.DIV_RelA_stats.945$SampLoc <- samp.df_945$SampLoc
ord.DIV_RelA_stats.945$sample_name <- samp.df_945$sample_name
ord.DIV_RelA_stats.945$Code<-paste(ord.DIV_RelA_stats.945$SampLoc,samp.df_945$Sub.loc)

ggplot(ord.DIV_RelA_stats.945, aes(x = NMDS1, y = NMDS2)) + 
  geom_point(aes(colour = Code), size = 7) +
  #scale_color_manual(values=Mcap.Parent.colors)+
  #geom_text(label = rownames(scores), nudge_x = 0.05, nudge_y = 0.05,check_overlap = T) +
  ggtitle("NMDS its2 Bray Curtis 945") +
  theme_classic() 
#ggsave("bc_all.pdf")

ggplot(ord.DIV_RelA_stats.945, aes(x = NMDS1, y = NMDS2)) + 
  geom_point(aes(colour = Code), size = 4,show.legend=T) +
  #scale_Dolor_manual(values=Sp.Dolors)+
  # geom_text(label = rownames(Clade), nudge_x = 0.05, nudge_y = 0.05,check_overlap = T) +
  # stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Code),show.legend=T) +
  geom_mark_ellipse(expand = 0,aes(fill=Code))+
  ggtitle("NMDS its2 945 only ") +
  theme_classic()
#ggsave("bc_all.pdf")





# for defense old code for code #### 

ggplot(scores.ord.DIV_RelA_stats, aes(x = NMDS1, y = NMDS2, color = Species, linetype=Treatment)) +
  geom_point(aes(shape=Treatment), size=4) +
  stat_ellipse(geom = "polygon",
               aes(fill = Code3), 
               alpha = 0.25)+
    theme_classic()

scores.ord.DIV_RelA_stats$Code4<-paste(scores.ord.DIV_RelA_stats$Species, scores.ord.DIV_RelA_stats$Clade, scores.ord.DIV_RelA_stats$Treatment)

 p<-ggplot(scores.ord.DIV_RelA_stats, aes(x = NMDS1, y = NMDS2, color = Species, linetype=Treatment)) +
  geom_point(aes(shape=Treatment), size=4) +
  stat_ellipse(geom = "polygon",
               aes(fill = Code4), 
               alpha = 0.25)+
    theme_classic()
p+facet_wrap(~Species)



#960 ####
DIV_RelA_stats_960<-subset_samples(coralDIV2.prune_stats, TagID=="960")
  DIV_RelA_stats_960 = prune_taxa(taxa_sums(DIV_RelA_stats_960) > 0, DIV_RelA_stats_960) #this removes any OTU with 0s
  samp.df_960 <- as(sample_data(DIV_RelA_stats_960), "data.frame") #sample df

#all speices
set.seed(30)

#make bray curtis data matrix
bc.960 <- phyloseq::distance(coralDIV2.prune_stats, method = "bray")
  samp.df_960 <- as(sample_data(coralDIV2.prune_stats), "data.frame") #sample df
#adonis test
adonis(bc.960 ~ SampLoc, data = samp.df_960)
    
# Homogeneity of dispersion test
# betaAll <- betadisper(bc.960, samp.df$Species)
# permutest(betaAll, pairwise = T) 
# (bc.all.HSD <- TukeyHSD(betaAll)) #do you need?
# plot(bc.all.HSD)

#Ordination NMDS for all species
set.seed(30)
ord.DIV_RelA_stats.960 <- ordinate(DIV_RelA_stats_960, "NMDS", "bray", trymax = 100) 
stressplot(ord.DIV_RelA_stats.960)
  samp.df_960 <- as(sample_data(DIV_RelA_stats_960), "data.frame") #sample df
ord.DIV_RelA_stats.960 <- as.data.frame(cbind(vegan::scores(ord.DIV_RelA_stats.960, display="sites"))) #make df
ord.DIV_RelA_stats.960$SampLoc <- samp.df_960$SampLoc
ord.DIV_RelA_stats.960$sample_name <- samp.df_960$sample_name
ord.DIV_RelA_stats.960$Location <- samp.df_960$Location

ord.DIV_RelA_stats.960$Code<-paste(ord.DIV_RelA_stats.960$SampLoc, ord.DIV_RelA_stats.960$Location)
ggplot(ord.DIV_RelA_stats.960, aes(x = NMDS1, y = NMDS2)) + 
  geom_point(aes(colour = Location), size = 7) +
  #scale_color_manual(values=Mcap.Parent.colors)+
  #geom_text(label = rownames(scores), nudge_x = 0.05, nudge_y = 0.05,check_overlap = T) +
  ggtitle("NMDS its2 Bray Curtis 960") +
  theme_classic() 

#ggsave("bc_all.pdf")
ggplot(ord.DIV_RelA_stats.960, aes(x = NMDS1, y = NMDS2)) + 
  geom_point(aes(colour = Land_Ocean, shape=Code), size = 4) +
  #scale_color_manual(values=Mcap.Parent.colors)+
  #geom_text(label = rownames(scores), nudge_x = 0.05, nudge_y = 0.05,check_overlap = T) +
  ggtitle("NMDS its2 Bray Curtis 960") +
  theme_classic() 
#ggsave("bc_all.pdf")

ggplot(ord.DIV_RelA_stats.960, aes(x = NMDS1, y = NMDS2)) + 
  geom_point(aes(colour = Code), size = 4,show.legend=T) +
  #scale_Dolor_manual(values=Sp.Dolors)+
  # geom_text(label = rownames(Clade), nudge_x = 0.05, nudge_y = 0.05,check_overlap = T) +
  # stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Code),show.legend=T) +
  geom_mark_ellipse(expand = 0,aes(fill=Code))+
  ggtitle("NMDS its2 960 only ") +
  theme_classic()
#ggsave("bc_all.pdf")

#970 ####
DIV_RelA_stats_970<-subset_samples(coralDIV2.prune_stats, TagID=="970")
  DIV_RelA_stats_970 = prune_taxa(taxa_sums(DIV_RelA_stats_970) > 0, DIV_RelA_stats_970) #this removes any OTU with 0s
  samp.df_970 <- as(sample_data(DIV_RelA_stats_970), "data.frame") #sample df

#all speices
set.seed(30)

#make bray curtis data matrix
bc.970 <- phyloseq::distance(coralDIV2.prune_stats, method = "bray")
  samp.df_970 <- as(sample_data(coralDIV2.prune_stats), "data.frame") #sample df
#adonis test
adonis(bc.970 ~ SampLoc, data = samp.df_970)
    
# Homogeneity of dispersion test
# betaAll <- betadisper(bc.all, samp.df$Species)
# permutest(betaAll, pairwise = T) 
# (bc.all.HSD <- TukeyHSD(betaAll)) #do you need?
# plot(bc.all.HSD)

#Ordination NMDS for all species
set.seed(30)
ord.DIV_RelA_stats.970 <- ordinate(DIV_RelA_stats_970, "NMDS", "bray", trymax = 100) 
stressplot(ord.DIV_RelA_stats.970)
  samp.df_970 <- as(sample_data(DIV_RelA_stats_970), "data.frame") #sample df

ord.DIV_RelA_stats.970 <- as.data.frame(cbind(vegan::scores(ord.DIV_RelA_stats.970, display="sites"))) #make df
ord.DIV_RelA_stats.970$SampLoc <- samp.df_970$SampLoc
ord.DIV_RelA_stats.970$sample_name <- samp.df_970$sample_name
ord.DIV_RelA_stats.970$Sub.loc <- samp.df_970$Sub.loc
ord.DIV_RelA_stats.970$Location<-samp.df_970$Location

ggplot(ord.DIV_RelA_stats.970, aes(x = NMDS1, y = NMDS2)) + 
  geom_point(aes(colour = Location), size = 7) +
  #scale_color_manual(values=Mcap.Parent.colors)+
  #geom_text(label = rownames(scores), nudge_x = 0.05, nudge_y = 0.05,check_overlap = T) +
  # geom_line(aes(group = Location))+
    geom_mark_ellipse(expand = 0,aes(fill=Location))+
  ggtitle("NMDS its2 Bray Curtis 970") +
  theme_classic() 
#ggsave("bc_all.pdf")

ggplot(ord.DIV_RelA_stats.970, aes(x = NMDS1, y = NMDS2)) + 
  geom_point(aes(colour = Location), size = 4,show.legend=T) +
  #scale_Dolor_manual(values=Sp.Dolors)+
  # geom_text(label = rownames(Clade), nudge_x = 0.05, nudge_y = 0.05,check_overlap = T) +
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Location),show.legend=T) +
  # geom_line(aes(group = Location),color="grey")+
  ggtitle("NMDS its2 970 only ") +
  theme_classic()
#ggsave("bc_all.pdf")



#966 ####
DIV_RelA_stats_966<-subset_samples(coralDIV2.prune_stats, TagID=="966")
  DIV_RelA_stats_966 = prune_taxa(taxa_sums(DIV_RelA_stats_966) > 0, DIV_RelA_stats_966) #this removes any OTU with 0s

#all speices
set.seed(30)

#make bray curtis data matrix
bc.966 <- phyloseq::distance(coralDIV2.prune_stats, method = "bray")
  samp.df_966 <- as(sample_data(coralDIV2.prune_stats), "data.frame") #sample df
#adonis test
adonis(bc.966 ~ SampLoc, data = samp.df_966)
    
#Ordination NMDS for all species
set.seed(30)
ord.DIV_RelA_stats.966 <- ordinate(DIV_RelA_stats_966, "NMDS", "bray", trymax = 100) 
stressplot(ord.DIV_RelA_stats.966)
  samp.df_966 <- as(sample_data(DIV_RelA_stats_966), "data.frame") #sample df
ord.DIV_RelA_stats.966 <- as.data.frame(cbind(vegan::scores(ord.DIV_RelA_stats.966, display="sites"))) #make df
ord.DIV_RelA_stats.966$SampLoc <- samp.df_966$SampLoc
ord.DIV_RelA_stats.966$sample_name <- samp.df_966$sample_name
ord.DIV_RelA_stats.966$Code <- paste(samp.df_966$SampLoc, samp.df_966$Sub.loc)


ggplot(ord.DIV_RelA_stats.966, aes(x = NMDS1, y = NMDS2)) + 
  geom_point(aes(colour = Code), size = 4) +
  #scale_color_manual(values=Mcap.Parent.colors)+
  #geom_text(label = rownames(scores), nudge_x = 0.05, nudge_y = 0.05,check_overlap = T) +
  ggtitle("NMDS its2 Bray Curtis 966") +
  theme_classic() 
#ggsave("bc_all.pdf")


ggplot(ord.DIV_RelA_stats.966, aes(x = NMDS1, y = NMDS2)) + 
  geom_point(aes(colour = Code), size = 4,show.legend=T) +
  #scale_Dolor_manual(values=Sp.Dolors)+
  # geom_text(label = rownames(Clade), nudge_x = 0.05, nudge_y = 0.05,check_overlap = T) +
  # stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Code),show.legend=T) +
  geom_mark_ellipse(expand = 0,aes(fill=Code))+
  ggtitle("NMDS its2 966 only ") +
  theme_classic()
#ggsave("bc_all.pdf")


#981 ####
DIV_RelA_stats_981<-subset_samples(coralDIV2.prune_stats, TagID=="981")
  DIV_RelA_stats_981 = prune_taxa(taxa_sums(DIV_RelA_stats_981) > 0, DIV_RelA_stats_981) #this removes any OTU with 0s

#all speices
set.seed(30)

#make bray curtis data matrix
bc.981 <- phyloseq::distance(coralDIV2.prune_stats, method = "bray")
  samp.df_981 <- as(sample_data(coralDIV2.prune_stats), "data.frame") #sample df
#adonis test
adonis(bc.981 ~ SampLoc, data = samp.df_981)


#Ordination NMDS for all species
set.seed(30)
ord.DIV_RelA_stats.981 <- ordinate(DIV_RelA_stats_981, "NMDS", "bray", trymax = 100) 
stressplot(ord.DIV_RelA_stats.981)
  samp.df_981 <- as(sample_data(DIV_RelA_stats_981), "data.frame") #sample df

ord.DIV_RelA_stats.981 <- as.data.frame(cbind(vegan::scores(ord.DIV_RelA_stats.981, display="sites"))) #make df
ord.DIV_RelA_stats.981$SampLoc <- samp.df_981$SampLoc
ord.DIV_RelA_stats.981$sample_name <- samp.df_981$sample_name
ord.DIV_RelA_stats.981$Location <- samp.df_981$Location

ggplot(ord.DIV_RelA_stats.981, aes(x = NMDS1, y = NMDS2)) + 
  geom_point(aes(colour = Location, shape=Location), size = 4) +
  #scale_color_manual(values=Mcap.Parent.colors)+
  #geom_text(label = rownames(scores), nudge_x = 0.05, nudge_y = 0.05,check_overlap = T) +
  ggtitle("NMDS its2 Bray Curtis 981") +
  theme_classic() 
#ggsave("bc_all.pdf")

ggplot(ord.DIV_RelA_stats.981, aes(x = NMDS1, y = NMDS2)) + 
  geom_point(aes(colour = Location), size = 4,show.legend=T) +
  #scale_Dolor_manual(values=Sp.Dolors)+
  # geom_text(label = rownames(Clade), nudge_x = 0.05, nudge_y = 0.05,check_overlap = T) +
  # stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Code),show.legend=T) +
  # geom_mark_ellipse(expand = 0,aes(fill=Location))+
    geom_mark_hull(expand = 0,aes(fill=Location))+

  ggtitle("NMDS its2 981 only ") +
  theme_classic()
#ggsave("bc_all.pdf")


#957 ####
DIV_RelA_stats_957<-subset_samples(coralDIV2.prune_stats, TagID=="957")
  DIV_RelA_stats_957 = prune_taxa(taxa_sums(DIV_RelA_stats_957) > 0, DIV_RelA_stats_957) #this removes any OTU with 0s

#all speices
set.seed(30)

#make bray curtis data matrix
bc.957 <- phyloseq::distance(coralDIV2.prune_stats, method = "bray")
  samp.df_957 <- as(sample_data(coralDIV2.prune_stats), "data.frame") #sample df
#adonis test
adonis(bc.957 ~ SampLoc, data = samp.df_957)


#Ordination NMDS for all species
set.seed(30)
ord.DIV_RelA_stats.957 <- ordinate(DIV_RelA_stats_957, "NMDS", "bray", trymax = 100) 
stressplot(ord.DIV_RelA_stats.957)
  samp.df_957 <- as(sample_data(DIV_RelA_stats_957), "data.frame") #sample df

ord.DIV_RelA_stats.957 <- as.data.frame(cbind(vegan::scores(ord.DIV_RelA_stats.957, display="sites"))) #make df
ord.DIV_RelA_stats.957$SampLoc <- samp.df_957$SampLoc
ord.DIV_RelA_stats.957$sample_name <- samp.df_957$sample_name
ord.DIV_RelA_stats.957$Location <- samp.df_957$Location

ggplot(ord.DIV_RelA_stats.957, aes(x = NMDS1, y = NMDS2)) + 
  geom_point(aes(colour = Location), size = 7) +
  #scale_color_manual(values=Mcap.Parent.colors)+
  #geom_text(label = rownames(scores), nudge_x = 0.05, nudge_y = 0.05,check_overlap = T) +
  ggtitle("NMDS its2 Bray Curtis 957") +
  theme_classic() 
#ggsave("bc_all.pdf")

ggplot(ord.DIV_RelA_stats.957, aes(x = NMDS1, y = NMDS2)) + 
  geom_point(aes(colour = Location), size = 4,show.legend=T) +
  #scale_Dolor_manual(values=Sp.Dolors)+
  # geom_text(label = rownames(Clade), nudge_x = 0.05, nudge_y = 0.05,check_overlap = T) +
  # stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Code),show.legend=T) +
  # geom_mark_ellipse(expand = 0,aes(fill=Location))+
    geom_mark_hull(expand = 0,aes(fill=Location))+

  ggtitle("NMDS its2 957 only ") +
  theme_classic()
#ggsave("bc_all.pdf")




#985 ####
DIV_RelA_stats_985<-subset_samples(coralDIV2.prune_stats, TagID=="985")
  DIV_RelA_stats_985 = prune_taxa(taxa_sums(DIV_RelA_stats_985) > 0, DIV_RelA_stats_985) #this removes any OTU with 0s

#all speices
set.seed(30)

#make bray curtis data matrix
bc.985 <- phyloseq::distance(coralDIV2.prune_stats, method = "bray")
  samp.df_985 <- as(sample_data(coralDIV2.prune_stats), "data.frame") #sample df
#adonis test
adonis(bc.985 ~ SampLoc, data = samp.df_985)


#Ordination NMDS for all species
set.seed(30)
ord.DIV_RelA_stats.985 <- ordinate(DIV_RelA_stats_985, "NMDS", "bray", trymax = 100) 
stressplot(ord.DIV_RelA_stats.985)
  samp.df_985 <- as(sample_data(DIV_RelA_stats_985), "data.frame") #sample df

ord.DIV_RelA_stats.985 <- as.data.frame(cbind(vegan::scores(ord.DIV_RelA_stats.985, display="sites"))) #make df
ord.DIV_RelA_stats.985$SampLoc <- samp.df_985$SampLoc
ord.DIV_RelA_stats.985$sample_name <- samp.df_985$sample_name
ord.DIV_RelA_stats.985$Location <- samp.df_985$Location

ggplot(ord.DIV_RelA_stats.985, aes(x = NMDS1, y = NMDS2)) + 
  geom_point(aes(colour = Location), size = 7) +
  #scale_color_manual(values=Mcap.Parent.colors)+
  #geom_text(label = rownames(scores), nudge_x = 0.05, nudge_y = 0.05,check_overlap = T) +
  ggtitle("NMDS its2 Bray Curtis 985") +
  theme_classic() 
#ggsave("bc_all.pdf")

ggplot(ord.DIV_RelA_stats.985, aes(x = NMDS1, y = NMDS2)) + 
  geom_point(aes(colour = Location), size = 4,show.legend=T) +
  #scale_Dolor_manual(values=Sp.Dolors)+
  # geom_text(label = rownames(Clade), nudge_x = 0.05, nudge_y = 0.05,check_overlap = T) +
  # stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Code),show.legend=T) +
  geom_mark_hull(expand = 0,aes(fill=Location))+
  ggtitle("NMDS its2 985 only ") +
  theme_classic()
#ggsave("bc_all.pdf")


#959 ####
DIV_RelA_stats_959<-subset_samples(coralDIV2.prune_stats, TagID=="959")
  DIV_RelA_stats_959 = prune_taxa(taxa_sums(DIV_RelA_stats_959) > 0, DIV_RelA_stats_959) #this removes any OTU with 0s

#all speices
set.seed(30)

#make bray curtis data matrix
bc.959 <- phyloseq::distance(coralDIV2.prune_stats, method = "bray")
  samp.df_959 <- as(sample_data(coralDIV2.prune_stats), "data.frame") #sample df
#adonis test
adonis(bc.959 ~ SampLoc, data = samp.df_959)


#Ordination NMDS for all species
set.seed(30)
ord.DIV_RelA_stats.959 <- ordinate(DIV_RelA_stats_959, "NMDS", "bray", trymax = 100) 
stressplot(ord.DIV_RelA_stats.959)
  samp.df_959 <- as(sample_data(DIV_RelA_stats_959), "data.frame") #sample df

ord.DIV_RelA_stats.959 <- as.data.frame(cbind(vegan::scores(ord.DIV_RelA_stats.959, display="sites"))) #make df
ord.DIV_RelA_stats.959$SampLoc <- samp.df_959$SampLoc
ord.DIV_RelA_stats.959$sample_name <- samp.df_959$sample_name
ord.DIV_RelA_stats.959$Sub.loc <- samp.df_959$Sub.loc
ord.DIV_RelA_stats.959$Code<-paste(ord.DIV_RelA_stats.959$SampLoc,ord.DIV_RelA_stats.959$Sub.loc)

ggplot(ord.DIV_RelA_stats.959, aes(x = NMDS1, y = NMDS2)) + 
  geom_point(aes(colour = Code), size = 7) +
  #scale_color_manual(values=Mcap.Parent.colors)+
  #geom_text(label = rownames(scores), nudge_x = 0.05, nudge_y = 0.05,check_overlap = T) +
  ggtitle("NMDS its2 Bray Curtis 959") +
  theme_classic() 
#ggsave("bc_all.pdf")

ggplot(ord.DIV_RelA_stats.959, aes(x = NMDS1, y = NMDS2)) + 
  geom_point(aes(colour = Sub.loc), size = 4,show.legend=T) +
  #scale_Dolor_manual(values=Sp.Dolors)+
  # geom_text(label = rownames(Clade), nudge_x = 0.05, nudge_y = 0.05,check_overlap = T) +
  # stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Code),show.legend=T) +
  geom_mark_hull(expand = 0,aes(fill=Sub.loc))+
  ggtitle("NMDS its2 959 only ") +
  theme_classic()
#ggsave("bc_all.pdf")


#961 ####
DIV_RelA_stats_961<-subset_samples(coralDIV2.prune_stats, TagID=="961")
  DIV_RelA_stats_961 = prune_taxa(taxa_sums(DIV_RelA_stats_961) > 0, DIV_RelA_stats_961) #this removes any OTU with 0s

#all speices
set.seed(30)

#make bray curtis data matrix
bc.961 <- phyloseq::distance(coralDIV2.prune_stats, method = "bray")
  samp.df_961 <- as(sample_data(coralDIV2.prune_stats), "data.frame") #sample df
#adonis test
adonis(bc.961 ~ SampLoc, data = samp.df_961)


#Ordination NMDS for all species
set.seed(30)
ord.DIV_RelA_stats.961 <- ordinate(DIV_RelA_stats_961, "NMDS", "bray", trymax = 100) 
stressplot(ord.DIV_RelA_stats.961)
  samp.df_961 <- as(sample_data(DIV_RelA_stats_961), "data.frame") #sample df

ord.DIV_RelA_stats.961 <- as.data.frame(cbind(vegan::scores(ord.DIV_RelA_stats.961, display="sites"))) #make df
ord.DIV_RelA_stats.961$SampLoc <- samp.df_961$SampLoc
ord.DIV_RelA_stats.961$sample_name <- samp.df_961$sample_name
ord.DIV_RelA_stats.961$Sub.loc <- samp.df_961$Sub.loc
ord.DIV_RelA_stats.961$Code<-paste(ord.DIV_RelA_stats.961$SampLoc,ord.DIV_RelA_stats.961$Sub.loc)

ggplot(ord.DIV_RelA_stats.961, aes(x = NMDS1, y = NMDS2)) + 
  geom_point(aes(colour = Code), size = 7) +
  #scale_color_manual(values=Mcap.Parent.colors)+
  #geom_text(label = rownames(scores), nudge_x = 0.05, nudge_y = 0.05,check_overlap = T) +
  ggtitle("NMDS its2 Bray Curtis 961") +
  theme_classic() 
#ggsave("bc_all.pdf")

ggplot(ord.DIV_RelA_stats.961, aes(x = NMDS1, y = NMDS2)) + 
  geom_point(aes(colour = Code), size = 4,show.legend=T) +
  #scale_Dolor_manual(values=Sp.Dolors)+
  # geom_text(label = rownames(Clade), nudge_x = 0.05, nudge_y = 0.05,check_overlap = T) +
  # stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = Code),show.legend=T) +
  geom_mark_hull(expand = 0,aes(fill=Code))+
  ggtitle("NMDS its2 961 only ") +
  theme_classic()
#ggsave("bc_all.pdf")




######### Try by tips and bases only #### 
coralDIV2.prune_statsBranch<-subset_samples(coralDIV2.prune_stats, SampLoc=="Branch")
coralDIV2.prune_statsBranch = prune_taxa(taxa_sums(coralDIV2.prune_statsBranch) > 0, coralDIV2.prune_statsBranch)
  
set.seed(10)
ord.DIV_bc.Branch <- ordinate(coralDIV2.prune_statsBranch, "NMDS", "bray", trymax = 500)  ### NO CONVERGENCE
stressplot(ord.DIV_bc.Branch)
  samp.df.Branch <- as(sample_data(coralDIV2.prune_statsBranch), "data.frame") #sample df

ord.DIV_bc.Branch <- as.data.frame(cbind(vegan::scores(ord.DIV_bc.Branch, display="sites"))) #make df
ord.DIV_bc.Branch$TagID <- samp.df.Branch$TagID
ord.DIV_bc.Branch$TagID<-as.factor(ord.DIV_bc.Branch$TagID)
ord.DIV_bc.Branch$SampLoc <- samp.df.Branch$SampLoc
ord.DIV_bc.Branch$Location<-samp.df.Branch$Sub.loc
ord.DIV_bc.Branch$Code<-paste(ord.DIV_bc.Branch$SampLoc, ord.DIV_bc.Branch$Location)
ord.DIV_bc.Branch$Code<-as.factor(ord.DIV_bc.Branch$Code)

ggplot(ord.DIV_bc.Branch, aes(x = NMDS1, y = NMDS2)) + 
  geom_point(aes(colour = TagID), size = 4,show.legend=T) +
  #scaleBrancholor_manual(values=Sp.Brancholors)+
  # geom_text(label = rownames(Clade), nudge_x = 0.05, nudge_y = 0.05,check_overlap = T) +
  stat_ellipse(aes(x = NMDS1, y = NMDS2, colour = TagID),show.legend=T) +
  ggtitle("NMDS its2 C only ") +
  themeBranchlassic()
#ggsave("bc_all.pdf")
```


